src/account/AccountGraphLoader.ts

```diff
@@ -155,12 +195,52 @@ export default class AccountsBatch {
         account.nextSettleTime,
         account.hasCashDebt,
         account.hasPortfolioAssetDebt,
-        Number(account.assetBitmapCurrency?.id),
-        account.balances.map(AccountsBatch.parseBalance),
-        account.portfolio.map(AccountsBatch.parseAsset),
+        account.assetBitmapCurrency?.id ? Number(account.assetBitmapCurrency.id) : undefined,
+        account.balances.map(AccountGraphLoader.parseBalance),
+        account.portfolio.map(AccountGraphLoader.parseAsset),
       );
       accounts.set(account.id, accountData);
     });
     return accounts;
   }
+
+  /**
+   * Returns a summary of an account's balances with historical transactions and internal return rate
+   */
+  public static async getBalanceSummary(address: string, accountData: AccountData, graphClient: GraphClient) {
+    const balanceHistory = await BalanceSummary.fetchBalanceHistory(address, graphClient);
+    const balanceSummary = BalanceSummary.build(accountData, balanceHistory);
+    return {balanceHistory, balanceSummary};
+  }
+
+  /**
+   * Returns the tradeHistory and assetSummary for an account
+   */
+  public static async getAssetSummary(address: string, accountData: AccountData, graphClient: GraphClient) {
+    const tradeHistory = await AssetSummary.fetchTradeHistory(address, graphClient);
+    const assetSummary = AssetSummary.build(accountData, tradeHistory);
+    return {tradeHistory, assetSummary};
+  }
+
+  /**
+   * Loads a single account
+   * @param graphClient
+   * @param address
+   * @returns AccountData instance for requested account
+   */
+  public static async load(graphClient: GraphClient, address: string) {
+    const {account} = await graphClient.queryOrThrow<AccountQueryResponse>(accountQuery, {id: address});
```

Comment:
@hboisselle I believe the graph always resolves addresses as lower case strings. (any upper case hex letters will not match), may want to confirm and update this accordingly.

Reply:
Confirmed subgraph only takes in lower case id.

Fixed the issue by always converting to lower case.